src/meta/proto-conv/tests/it/v025_user_stage.rs

```diff
@@ -0,0 +1,81 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_meta_types as mt;
+use common_meta_types::UserIdentity;
+use common_storage::StorageFsConfig;
+use common_storage::StorageParams;
+
+use crate::common;
+
+// These bytes are built when a new version in introduced,
+// and are kept for backward compatibility test.
+//
+// *************************************************************
+// * These messages should never be updated,                   *
+// * only be added when a new version is added,                *
+// * or be removed when an old version is no longer supported. *
+// *************************************************************
+//
+// The message bytes are built from the output of `test_user_stage_fs_latest()`
+#[test]
+fn test_decode_v25_user_stage() -> anyhow::Result<()> {
+    let user_stage_info_v25 = vec![
+        10, 17, 102, 115, 58, 47, 47, 100, 105, 114, 47, 116, 111, 47, 102, 105, 108, 101, 115, 26,
+        25, 10, 23, 18, 21, 10, 13, 47, 100, 105, 114, 47, 116, 111, 47, 102, 105, 108, 101, 115,
+        160, 6, 25, 168, 6, 24, 34, 37, 8, 1, 16, 128, 8, 26, 1, 124, 34, 2, 47, 47, 40, 2, 50, 1,
+        92, 58, 3, 114, 111, 119, 66, 3, 78, 97, 78, 74, 2, 39, 39, 160, 6, 25, 168, 6, 24, 42, 9,
+        10, 2, 48, 2, 16, 142, 8, 24, 1, 50, 4, 116, 101, 115, 116, 56, 100, 66, 29, 10, 8, 100,
+        97, 116, 97, 98, 101, 110, 100, 18, 11, 100, 97, 116, 97, 98, 101, 110, 100, 46, 114, 115,
+        160, 6, 25, 168, 6, 24, 160, 6, 25, 168, 6, 24,
+    ];
+
+    let want = mt::UserStageInfo {
+        stage_name: "fs://dir/to/files".to_string(),
+        stage_type: mt::StageType::LegacyInternal,
+        stage_params: mt::StageParams {
+            storage: StorageParams::Fs(StorageFsConfig {
+                root: "/dir/to/files".to_string(),
+            }),
+        },
+        file_format_options: mt::FileFormatOptions {
+            format: mt::StageFileFormatType::Json,
+            skip_header: 1024,
+            field_delimiter: "|".to_string(),
+            record_delimiter: "//".to_string(),
+            nan_display: "NaN".to_string(),
+            compression: mt::StageFileCompression::Bz2,
+            escape: "\\".to_string(),
+            row_tag: "row".to_string(),
+            quote: "\'\'".to_string(),
+        },
+        copy_options: mt::CopyOptions {
+            on_error: mt::OnErrorMode::AbortNum(2),
+            size_limit: 1038,
+            split_size: 0,
+            purge: true,
+            single: false,
+            max_file_size: 0,
+        },
+        comment: "test".to_string(),
+        number_of_files: 100,
+        creator: Some(UserIdentity {
+            username: "databend".to_string(),
+            hostname: "databend.rs".to_string(),
+        }),
+    };
+    common::test_load_old(func_name!(), user_stage_info_v25.as_slice(), 25, want)?;

```

Comment:
Add this test too, it tests encoding/decoding by the program of the latest version:
```
    common::test_pb_from_to(func_name!(), want())?;

```

https://github.com/datafuselabs/databend/blob/78d127257519bea475ad1361ee0359da764a443b/src/meta/proto-conv/tests/it/v024_table_meta.rs#L148